packages/aws-cdk-lib/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -460,11 +507,17 @@ export class StateMachine extends StateMachineBase {
       tracingConfiguration: props.tracingEnabled ? this.buildTracingConfiguration() : undefined,
       ...definitionBody.bind(this, this.role, props, graph),
       definitionSubstitutions: props.definitionSubstitutions,
+      encryptionConfiguration: props.kmsKey? {
+        kmsKeyId: props.kmsKey.keyArn,
+        kmsDataKeyReusePeriodSeconds: props.kmsDataKeyReusePeriodSeconds? props.kmsDataKeyReusePeriodSeconds.toSeconds() : 300,
+        type: 'CUSTOMER_MANAGED_KMS_KEY',
+      }: {
+        type: 'AWS_OWNED_KEY',
```

Comment:
Do `kmsKeyId` and `kmsDataKeyReusePeriodSeconds` only apply for customer managed keys?

Reply:
Yes only for customer managed keys